src/main/kotlin/com/redhat/devtools/intellij/kubernetes/editor/ResourceFile.kt

```diff
@@ -10,229 +10,172 @@
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.openapi.fileTypes.FileTypeRegistry
 import com.intellij.openapi.util.io.FileUtil
 import com.intellij.openapi.util.io.FileUtilRt
-import com.intellij.openapi.vfs.LocalFileSystem
 import com.intellij.openapi.vfs.VfsUtil
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.util.ui.EdtInvocationManager
 import com.redhat.devtools.intellij.common.editor.AllowNonProjectEditing
 import com.redhat.devtools.intellij.common.utils.UIHelper
+import com.redhat.devtools.intellij.kubernetes.model.Notification
+import com.redhat.devtools.intellij.kubernetes.model.util.createResource
+import com.redhat.devtools.intellij.kubernetes.model.util.trimWithEllipsis
 import io.fabric8.kubernetes.api.model.HasMetadata
-import io.fabric8.kubernetes.api.model.Namespace
+import io.fabric8.kubernetes.api.model.KubernetesResource
 import io.fabric8.kubernetes.client.utils.Serialization
-import io.fabric8.openshift.api.model.Project
 import org.apache.commons.io.FileUtils
+import org.apache.commons.io.IOUtils
 import org.jetbrains.yaml.YAMLFileType
+import java.io.IOException
+import java.nio.charset.Charset
 import java.nio.charset.StandardCharsets
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
 import java.util.function.Supplier
 
 open class ResourceFile protected constructor(
-    open val path: Path,
-    protected open var _virtualFile: VirtualFile? = null
+    open val virtualFile: VirtualFile
 ) {
 
     companion object Factory {
         const val EXTENSION = YAMLFileType.DEFAULT_EXTENSION
+
         @JvmStatic
-        val TEMP_FOLDER: Path = Paths.get(FileUtil.resolveShortWindowsName(FileUtil.getTempDirectory()), "intellij-kubernetes")
+        val TEMP_FOLDER: Path = Paths.get(FileUtil.resolveShortWindowsName(FileUtil.getTempDirectory()))
 
         @JvmStatic
         fun create(resource: HasMetadata): ResourceFile? {
-            val path = getPathFor(resource)
-            return create(path)
+            val virtualFile = createTemporaryFile(resource) ?: return null
+            return ResourceFile(virtualFile)
         }
 
         @JvmStatic
         fun create(virtualFile: VirtualFile?): ResourceFile? {
-            if (virtualFile == null) {
+            if (virtualFile == null
+                || !isResourceFile(virtualFile)) {
                 return null
             }
-            return create(VfsUtil.virtualToIoFile(virtualFile).toPath(), virtualFile)
-        }
-
-        @JvmStatic
-        fun create(path: Path, virtualFile: VirtualFile? = null): ResourceFile? {
-            if (!isResourceFile(path)) {
-                return null
+            return ResourceFile(virtualFile)
+        }
+
+        private fun createTemporaryFile(resource: HasMetadata): VirtualFile? {
+            return try {
+                val file = FileUtilRt.createTempFile(resource.metadata.name, ".$EXTENSION")
```

Comment:
Not sure if this is the behavior you want but let's say i open twice the same resource from the cluster, it will open two tabs with the same name. The only difference will be the tempfile because here it will be created a new one every call.
![image](https://user-images.githubusercontent.com/49404737/137185545-b058b0b4-8622-4f7d-b9f3-04cbadf262c5.png)